### DISCUSSION TITLE
[Feature] Shared Albums Photo Show On Shared User's timeline.

### DISCUSSION THREAD

**Discussion by mutuyihao** [immich-app/immich#7047] [Feature Request]
### The feature

I hope there is a toggle to detemined to show shared photos on timeline in ios app/ web /android of shared user account settings. And also, I have a question about How many people can login in a same time and keep login status by using a same account include ios app/ web /android.  

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by aisbergde** [comment]
Showing all shared assets would messing up my timeline, adding only selected would be better (for me). That's why I would like to modify / extend the request:
in a shared album I would like to "import" (virtually or real) selected assets into my "collection" and then these imported assets should be shown in my timeline.

**Comment by ttyridal** [comment]
When "partner sharing" you already have that option for web at least. (Account->Settings->Sharing)  There is a toggle if you want to see the shared photos in your timeline or not.

**Reply by ejpenney** [reply]
This will add a partner's images to the timeline, but not shared albums. I know I would like the option to add a shared album to my timeline.

**Comment by TheGorf** [comment]
I just want to be able to have one place to go to see albums, shared or not. In the time it took me to type this, both my wife and my parents have complained about going to multiple places to see it. We really want to just see "Albums" as in my albums and shared albums. Since theres no difference, why require going to two different places? Current layout is kinda dumb in my opinion.

**Comment by aviv926** [comment]
Related https://github.com/immich-app/immich/discussions/1587

**Comment by tbleiker** [comment]
I would like to see this as well!

Use case: Share multiple albums (e.g. fotos of your Kids or from family meetings with family members). You create an account for each and setup the mobile app. Now they can see the shared albums, but have to switch between them. It would be nice if the fotos of all shared albums (or selected ones) are shown in the timeline.

**Comment by MadeInPierre** [comment]
+1 This is necessary in a family server. 

My use case: I don't want to merge the full timelines from all my family members (i.e. partner sharing) as they have photos from their own activities without me. Our current solution is to share albums with photos from events when we're together. However, their photos only show inside these albums and not on my timeline, which is sad as the ultimate goal is that my timeline should show all photos related to me.

**Comment by ty-han** [comment]
+1  - I think this should be three options here:

1) Add all shared albums to timeline (from settings).  If they have not enabled this, then they'd be presented with these two options:
2) Add an entire shared album to timeline, if 100% is relevant to me. 
3) Add specific photos from shared album to timeline if only X% are relevant to me.

**Reply by mandalarigabriele** [reply]
totally agree! Point 1 is really helpful!

**Reply by smartVanture** [reply]
Point 2 and.3 is pretty much as Google Photos does it, i think.

**Comment by ZenomeFR** [comment]
Hi, I'd also like to see this option in the future. In my case, my wife and I have our own account in my server in order to not mix our stuff but during holidays we create a shared album and put all of our photos inside it. It would be nice to have an option to "add to my timeline" on each album.

Thanks :)

**Reply by MadeInPierre** [reply]
Exactly, I have the exact same problem with my server of 4 family members :)

**Reply by ZenomeFR** [reply]
+ I've also discover that shared pictures are not included in face recognition (I can't add a face if I don't "own" the picture) & I can't add tags nor stars-rating to those shared pictures :(

**Comment by baozi510** [comment]
i need this.
my mom can't find my shared photos.because need to switch tabs, she's old and not good with smartphones
my temporary solution runs a script to upload the shared photos to her account timeline

**Reply by S-Pika** [reply]
Thinking about doing this myself for the time being until this type of feature is added. Did you find a good system to deal with the duplicate files or are you just accepting the duplicate storage cost of having the images in both accounts?

My idea was to create a Family account in immich and then have a script download all the photos from a shared album then upload it to that family account and then add that family account as a partner to all the people who want the shared photos in their timeline.

I am running in to an issue of the duplicate files obviously and also an issue with the thumbnails for the photos in the family account not properly displaying.

**Comment by bo0tzz** [comment]
#3984

**Comment by carlolars** [comment]
I'd like to have an option to partner sharing that instead of adding *all* photos from my partner to my timeline only photos in albums shared by my partner to me should be added to my timeline.
This way my timeline won't get cluttered by my wifes irrelevant photos and she can take whatever photos she wants without having to think that this will show up on my timeline.

**Reply by carlolars** [reply]
I'm sorry but I posted this in the wrong discussion (to many tabs :smile: ), I reposted in the intended place https://github.com/immich-app/immich/discussions/1779#discussioncomment-13867645

**Comment by jmvgmr** [comment]
I would like to have this feature as well

**Comment by srvprivate** [comment]
It looks like I am discovering a bug potentially along these lines. All of what they've said is awesome, in addition to that, it seems the logic of Shared Albums to a Shared Partner do not give the option to show in timeline. So if a Partner has been invited to view said album, it does not show on timeline, but if partner who was invited uploads to that album, it appears both parties see it.

**Reply by srvprivate** [reply]
I also made the same mistake of posting here, #1779 (comment)

**Comment by Antiheld86** [comment]
Hello,

I forked the repo and builded a branch with the feature to enable shared albums in the timeline [here](https://github.com/Antiheld86/immich/tree/feature/shared-albums-in-timeline). If someone is interested you can test it. I also send a pull request #23660. I am thinking about making it possible to enable it on a per album basis.

It is also not yet implemented in the mobile app.

**Reply by Antiheld86** [reply]
It is now also possible to decide which shared album to show.

**Reply by ty-han** [reply]
Thank you!  I would help if I could code. This is actually a major blocker from me using the app, as I do not want to enable partner sharing with my family, but also do want them to see shared albums in their timeline.  Please definitely keep us updated with your progress!

**Reply by Antiheld86** [reply]
The feature is fully implemented in the Web-App. If you want to use it you can do it. I try my best to keep the repo updated with commits from the official repo. There is also a pre-build docker-container.

`ghcr.io/antiheld86/immich-server:latest`

 For now I wont implement the feature in the mobile app.

My pull-request got rejected because of #12614 so it wont be available in the official repo.

**Comment by JosiasDMartins** [comment]
It makes total sense. We have a newborn at home and I added my mother and mother in law to out ImMich server. They would like to have all photos we add on the baby's album on there timeline, since it's easier to track new photos and see them, however there are no options for doing it. It would be great if there are a button inside the shared album to mark it as "add to timeline".

**Reply by devonverdonk** [reply]
Totally agree, I really hope this will reach the roadmap someday
